### DISCUSSION TITLE
Question about generating XML files through the python script.

### DISCUSSION THREAD

**Discussion by Ethan-xj** [neams-th-coe/cardinal#866] [Q&A] [answered]
@aprilnovak Hello! I've been encountering a common issue recently while testing OpenMC functionality, where I'm unable to generate the corresponding XML files using a Python script. 
Take the `/turotials/gas_assembly` for example, when I run the command: `python assembly.py`, there is an error: 
```
(base) ethan@ethan-pc:~/cardinal/tutorials/gas_assembly$ python assembly.py
Volume of fuel bundle (m3):  0.7276669595808495
Volume of solid regions in fuel bundle (m3):  0.5899306909225606
Traceback (most recent call last):
  File "/home/ethan/cardinal/tutorials/gas_assembly/assembly.py", line 433, in <module>
    main()
  File "/home/ethan/cardinal/tutorials/gas_assembly/assembly.py", line 429, in main
    model = assembly(args.n_axial, args.n_inactive, args.n_active, args.entropy)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ethan/cardinal/tutorials/gas_assembly/assembly.py", line 330, in assembly
    hex_prism = openmc.hexagonal_prism(bundle_pitch / math.sqrt(3.0), 'x', boundary_type='periodic')
                ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'openmc' has no attribute 'hexagonal_prism'
```
I think it's because the version of openmc changes. Furthermore, there have been some changes in syntax. Here is the version of my openmc: `0.14.1-dev`
And after replacing syntax `hexagonal_prism` with `openmc.model.HexagonalPrism` as indicated in the manual, I encountered additional errors:
```
(base) ethan@ethan-pc:~/cardinal/tutorials/gas_assembly$ python assembly.py
Volume of fuel bundle (m3):  0.7276669595808495
Volume of solid regions in fuel bundle (m3):  0.5899306909225606
Traceback (most recent call last):
  File "/home/ethan/cardinal/tutorials/gas_assembly/assembly.py", line 433, in <module>
    main()
  File "/home/ethan/cardinal/tutorials/gas_assembly/assembly.py", line 429, in main
    model = assembly(args.n_axial, args.n_inactive, args.n_active, args.entropy)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ethan/cardinal/tutorials/gas_assembly/assembly.py", line 331, in assembly
    outer_cell = openmc.Cell(region=hex_prism & +min_z & -max_z, fill=hex_lattice)
                                    ~~~~~~~~~~^~~~~~~~
TypeError: unsupported operand type(s) for &: 'HexagonalPrism' and 'Halfspace'
```
It's a little bit difficult for me to modify the script. What should I do?

Ethan

**Comment by aprilnovak** [comment, marked as answer]
Ah, sorry about that! We don't test the Python scripts in our regression suite, so these fell a bit out of date. You were most of the way there with the fix, but the changes in OpenMC also changed the underlying "type" of the object being created (from a `region` to a `surface`, which you now need to use `+`/`-` to indicate the half-spaces for).

[I've fixed this here](https://github.com/neams-th-coe/cardinal/pull/867), which I'll merge as soon as the tests pass (~1 hour). Once that is merged, you can then pull in the latest Cardinal and everything should work.

Thanks for pointing this out!

**Reply by Ethan-xj** [reply]
Thank you so much for your help!
